sdk/core/core/inc/az_http_request_builder_curl.h

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_HTTP_REQUEST_BUILDER_CURL_H
+#define AZ_HTTP_REQUEST_BUILDER_CURL_H
+
+#include <az_http_request_builder.h>
+
+#include <az_curl_adapter.h>
+
+#include <_az_cfg_prefix.h>
+
+az_result az_http_request_builder_to_curl(
+    az_http_request_builder const * const p_builder,
```

Comment:
I don't understand this function. Actually the whole header file (and it's c file).
what are we expecting with this one?  it is taking a builder as parameter? why?

I believe that by the time a request will be translated to curl, it has been built already. Then, the adapter would only receive the buffer with the request and another buffer for writing the response in it. 

Probably we don't really need this az_http_request_builder_curl.h/c